controllers/argocd/argocd_controller_test.go

```diff
@@ -102,6 +102,151 @@ func TestReconcileArgoCD_Reconcile(t *testing.T) {
+	}
+
+	//Instance 'c'
+	req3 := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      c.Name,
+			Namespace: c.Namespace,
+		},
+	}
+	res3, err := rt.Reconcile(context.TODO(), req3)
+	assert.NoError(t, err)
+	if res3.Requeue {
+		t.Fatal("reconcile requeued request")
+	}
+
+	// Apply label-selector foo=bar to the operator but not to the argocd instances. No reconciliation will happen and an error is expected.
+	rt.LabelSelector = "foo=bar"
+	reqTest := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      a.Name,
+			Namespace: a.Namespace,
+		},
+	}
+	resTest, err := rt.Reconcile(context.TODO(), reqTest)
+	assert.Error(t, err)
+	if resTest.Requeue {
+		t.Fatal("reconcile requeued request")
+	}
+
+	//not reconciled should return error
+	reqTest2 := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      b.Name,
+			Namespace: b.Namespace,
+		},
+	}
+	resTest2, err := rt.Reconcile(context.TODO(), reqTest2)
+	assert.Error(t, err)
+	if resTest2.Requeue {
+		t.Fatal("reconcile requeued request")
+	}
+
+	//not reconciled should return error
+	reqTest3 := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      c.Name,
+			Namespace: c.Namespace,
+		},
+	}
+	resTest3, err := rt.Reconcile(context.TODO(), reqTest3)
+	assert.Error(t, err)
+	if resTest3.Requeue {
+		t.Fatal("reconcile requeued request")
+	}
+}
+func TestReconcileArgoCD_ReconcileLabel(t *testing.T) {
+
+	// Multiple ArgoCD instances present with matching label present on some and absent on some.
+	// Only instances matching the label are reconciled.
+
+	logf.SetLogger(ZapLogger(true))
+	ctx := context.Background()
+	a := makeTestArgoCD()
+	r1 := makeTestReconciler(t, a)
+	assert.NoError(t, createNamespace(r1, a.Namespace, ""))
+	b := makeTestArgoCD()
+	r2 := makeTestReconciler(t, b)
+	assert.NoError(t, createNamespace(r2, b.Namespace, ""))
+
+	// Apply label foo=bar to the argocd instance and to the operator for reconciliation without any error.
+	r1.LabelSelector = "foo=bar"
+	r2.LabelSelector = "foo=bar"
+
+	//Apply label to Instance 'a' but not to Instance 'b'
+	a.SetLabels(map[string]string{"foo": "bar"})
+	err := r1.Client.Update(ctx, a)
+	fatalIfError(t, err, "failed to update the ArgoCD: %s", err)
+	reqTest2 := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      a.Name,
+			Namespace: a.Namespace,
+		},
+	}
+	resTest2, err := r1.Reconcile(context.TODO(), reqTest2)
+
+	//Instance 'a' reconciled without error
+	assert.NoError(t, err)
+	if resTest2.Requeue {
+		t.Fatal("reconcile requeued request")
+	}
+
+	reqTest3 := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      b.Name,
+			Namespace: b.Namespace,
+		},
+	}
+	resTest3, err := r2.Reconcile(context.TODO(), reqTest3)
+	//Instance 'b' is not reconciled and an error is expeced
+	assert.Error(t, err)
+	if resTest3.Requeue {
+		t.Fatal("reconcile requeued request")
+	}
+
+}
+
```

Comment:
@raghavi101 @anandf do we need 2 separate tests for this?
I feel like this could be handled in a single combined  test case:
- create multiple instances - some with no label, others with 2 or more different labels and no label filter set
- ensure all instances reconcile
- add a labelselector that matches the labels on one of the existing instances we just made, re-reconcile all of them and now only  the one with the matching label should not error out

Reply:
also its a little cleaner to do 
```
a := makeTestArgoCD( func( a *argoproj.ArgoCD) {
          a. Name = "argo-test-1"
          a.Labels["some-label"] = "some val"
          })
```
instead of
```
a := makeTestArgoCD()
a.Name = "argo-test-1"
```

Reply:
Combined both the test cases, thanks @jaideepr97